crates/languages/src/rust.rs

```diff
@@ -792,23 +808,24 @@ fn package_name_and_bin_name_from_abs_path(
 
     let metadata: CargoMetadata = serde_json::from_slice(&output).log_err()?;
 
-    retrieve_package_id_and_bin_name_from_metadata(metadata, abs_path).and_then(
-        |(package_id, bin_name, target_kind)| {
+    retrieve_package_id_and_bin_info_from_metadata(metadata, abs_path).and_then(
+        |(package_id, bin_name, required_features, target_kind)| {
             let package_name = package_name_from_pkgid(&package_id);
 
             package_name.map(|package_name| TargetInfo {
                 package_name: package_name.to_owned(),
                 target_name: bin_name,
+                required_features,
                 target_kind,
             })
         },
     )
 }
 
-fn retrieve_package_id_and_bin_name_from_metadata(
+fn retrieve_package_id_and_bin_info_from_metadata(
     metadata: CargoMetadata,
     abs_path: &Path,
-) -> Option<(String, String, TargetKind)> {
```

Comment:
> -> Option<(String, String, Option<Vec<String>>, TargetKind)>

This is `Option<TargetInfo>` it seems, and we can name the function correspondingly (maybe, even turn this into `new`?)

> Option<Vec<String>

Can it be just `Vec<String>` with `#[serde(default)]`?

Reply:
> This is Option<TargetInfo> it seems, and we can name the function correspondingly (maybe, even turn this into new?)

While the types look the same, `TargetInfo` contains the package name and this method returns the package id.

> Can it be just Vec<String> with #[serde(default)]?
done :+1:

Reply:
I refactored the method to no longer return the tuple